### ISSUE TITLE
Selectively optimize acquisition function over a subset of input domain

### ISSUE THREAD

**Issue by g1Jeong** [SheffieldML/GPyOpt#204]
Hello,
I'm afraid that my original question was too wordy, and so here I put it much more simply :

Long story short, **is there a way to selectively optimize an acquisition function over a subset of variables, instead of over the whole variables?** 

Let's say a GP model is defined with a multidimensional input variable x = (x1, x2). However, the acquisition is defined only over x1, and hence the optimization should only take x1 into account. How can I implement this??

Here's the part of the acquisition function I'm trying to design, 
```    def _get_mean_cov(self, x):
        # x : r x 1 vector
        X = self.get_X(x, self.ys.shape[0])
        Xy = np.concatenate((X, self.ys), axis = 1)
        mu, cov = self.model.model.predict(Xy, full_cov=True)
        mu = mu - self.T
        return mu, cov

    @staticmethod
    def get_X(x, n):
        x = x.reshape(max(x.shape), 1)
        X = np.matmul(np.ones((n, x.shape[0])), x)
        return X
```
As can be seen from the code above, the model is taking self.ys as well as x; however, self.ys is a fixed vector, thus making the optimization of the acquisition only over x. 
(This code is just a crude implementation, being presented here only to show a glimpse of what I'm trying to do.)

Any comments and helps will be appreciated! 

> (The original question)
> I am trying to define a new acquisition function, and in the problem I'm dealing with, I need to optimize the acquisition over only a subset of input dimensions, not all. In other words, the GP model should be trained over the entire input dimensions; however, for a certain set of dimensions, available values are preset and are not changing, rendering the optimization over the entire input dimensions useless. The sweep over non-optimizable dimensions will be too costly, when there are only a few available values in fact. 
> 
> More concretely, if the input variable can be denoted as (x, y), where x is n dimensional and y is m dimensional, only n dimensional variable x is optimizable while y is not. And we know that y will have a fixed finite number of values (y1, y2, ..., y10(say) ). This means that the acquisition function should be defined only over x, not y, while incorporating some information that the model would give by considering the available y values.
> 
> Is there a simple way to implement this? I can code the acquisition part for it to receive a variable only of optimizable dimensions, but I'm afraid that the optimizer will anyway use the entire input domain (including where y is defined). May be I have to carefully follow how the acquisition optimizer is defined, but it doesn't seem to be so simple.. 
> 
> Can anyone have any idea regarding this ??

**Comment by apaleyes** [comment]
Yes, this is certainly possible. GPyOpt has a notion of context: you fix some variables to some values, and optimize over the rest. Example can be seen in this notebook: https://nbviewer.jupyter.org/github/SheffieldML/GPyOpt/blob/master/manual/GPyOpt_context.ipynb
